gateway/mw_auth_key.go

```diff
@@ -94,11 +92,6 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 		return nil, http.StatusOK
 	}
 
-	// skip auth key check if the request is looped.
```

Comment:
I don't see why we would. The Auth config changes for ApiB, internal request would still need to pass through auth mw; So we maybe do auth in several middlewares and *none* of them have ctxCheckLimits checks.

Reply:
skip auth key check if it is **self** looped. 
we had to distinguish it early on